tests/test_recipient_csv.py

```diff
@@ -738,16 +739,18 @@ def test_bad_or_missing_data(
             +2304031000, Mauritius
         """,
             set(),
+            3,
         ),
     ],
 )
-def test_international_recipients(file_contents, rows_with_bad_recipients):
+def test_international_recipients(file_contents, rows_with_bad_recipients, expectected_count):
     recipients = RecipientCSV(
         file_contents,
         template=_sample_template("sms"),
         allow_international_sms=True,
     )
     assert _index_rows(recipients.rows_with_bad_recipients) == rows_with_bad_recipients
+    assert recipients.international_sms_count == expectected_count
```

Comment:
This assertion only passes if the line above is run. Otherwise `international_sms_count` is always `0`. This feels fragile.

If you look at how this works for the existing limit then the class has a property which returns `True`/`False` and does the counting only when it’s needed:
https://github.com/alphagov/notifications-utils/blob/bf94e62d250b12fa0e5dfe50e50ff06ae5f831eb/notifications_utils/recipients.py#L189-L191

Reply:
Great catch, thank you! I made an update that fixes this issue, and moved the assertion up in the test.